ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1703,6 +1703,126 @@ class EclCmdStatus : public EclCmdCommon
     unsigned int       optListLimit;
 };
 
+class EclCmdZapGen : public EclCmdCommon
+{
+public:
+    EclCmdZapGen()
+    {
+        optObj.accept = eclObjWuid;
+    }
+    virtual eclCmdOptionMatchIndicator parseCommandLineOptions(ArgvIterator &iter)
+    {
+        eclCmdOptionMatchIndicator retVal = EclCmdOptionNoMatch;
+        if (iter.done())
+            return EclCmdOptionNoMatch;
+
+        for (; !iter.done(); iter.next())
+        {
+            const char *arg = iter.query();
+            if (iter.matchOption(optWuid, ECLOPT_WUID)||iter.matchOption(optWuid, ECLOPT_WUID_S))
+            {
+                optObj.type = eclObjWuid;
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optPath, ECLOPT_PATH))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optIncThorSlave, ECLOPT_INC_THOR_SLAVE_LOGS))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            if (iter.matchOption(optProblemDesc, ECLOPT_PROBLEM_DESC))
+            {
+                retVal = EclCmdOptionMatch;
+                continue;
+            }
+            eclCmdOptionMatchIndicator ind = EclCmdCommon::matchCommandLineOption(iter, true);
+            if (ind != EclCmdOptionMatch)
+                return ind;
+        }
+        return retVal;
+    }
+    virtual bool finalizeOptions(IProperties *globals)
+    {
+        if (!EclCmdCommon::finalizeOptions(globals))
+            return false;
+        return true;
+    }
+    virtual int processCMD()
+    {
+        if (optWuid.isEmpty())
```

Comment:
This kind of check is usually in "finalizeOptions()", which happens before processCMD().

Also should print to stderr not stdout.

Reply:
Changed, moved